client/components/domains/domain-registration-suggestion/index.jsx

```diff
@@ -28,32 +28,42 @@ const DomainRegistrationSuggestion = React.createClass( {
 
 	render() {
 		const { suggestion, translate } = this.props,
-			isAdded = hasDomainInCart( this.props.cart, suggestion.domain_name ),
+			domain = suggestion.domain_name,
+			isAdded = hasDomainInCart( this.props.cart, domain ),
 			domainFlags = [];
+
 		let buttonClasses, buttonContent;
 
-		if ( suggestion.domain_name ) {
+		if ( domain ) {
 			const newTLDs = [];
+			const a8cTLDs = [ '.ca', '.de', '.fr' ];
+			const tld = domain.substring( domain.indexOf( '.' ) );
```

Comment:
The old code handled `co.uk` too, by checking that there's no dot in the "left over" part of the domain [#41](https://github.com/Automattic/wp-calypso/pull/13485/files/bc645b1b5f14bfb1b3c9cfeb7732b40161777881#diff-d94c498e4ad71ff0c5e5bb7fc384bc99L41). I was aiming for simplifying it a little, but preserving the logic.

I share @klimeryk's mixed feelings about subdomains. I think for now I'll add a comment here to make it clearer why we're doing it this way and also mention subdomains.

```node
> suggestion = { domain_name: "example.co.uk" }
{ domain_name: 'example.co.uk' }
> tld = "co.uk"
'co.uk'
> suggestion.domain_name.substring( 0, suggestion.domain_name.length - ( tld.length + 1 ) )
'example'
```